components/Link/index.js

```diff
@@ -0,0 +1,242 @@
+	color: var(--wp--style--color--link, var(--global--color-primary));
+	position: relative;
+	display: inline-flex;
+	align-items: center;
+	gap: 0.5em;
+
+	/* This  holds the text URL input */
+	& > div {
+		text-decoration: underline;
+	}
+
+	.dashicon {
+		text-decoration: none;
+		font-size: 1em;
+		width: 1.5em;
+		height: 1.5em;
+		border-radius: 50%;
+		background: transparent;
+		display: flex;
+		align-items: center;
+		justify-content: center;
+		color: var(--color--warning);
+	}
+`;
+
+/**
+ * Link component that can be used inside other Gutenberg blocks for setting up URLs.
+ *
+ * The link should not be visible if the block is not focused. This will mainain nicer
+ * visuals in the block editor as a whole.
+ *
+ * @param {object} props								All properties passed to the component.
+ * @param {string} props.value 							The text to show inside the link
+ * @param {string} props.type 							Post or Page, used to autosuggest content for URL
+ * @param {boolean} props.opensInNewTab 				Should the link open in a new tab?
+ * @param {string} props.url 							The actual link to be set as href
+ * @param {Function} props.onLinkChange 				Callback when the URL is changed
+ * @param {Function} props.onTextChange 				Callback when the link's text is changed
+ * @param {string} props.kind 							Page or Post
+ * @param {string} props.placeholder 					Text visible before actual value is inserted
+ *
+ * @returns {JSX}
+ */
+const Link = ({
+	value,
+	type,
+	opensInNewTab,
+	url,
+	onLinkChange,
+	onTextChange,
+	kind,
+	placeholder,
+}) => {
+	const ref = useRef();
+	const [isLinkOpen, setIsLinkOpen] = useState(false);
+	const [isValid, setIsValid] = useState(false);
+	const { isSelected } = useBlockEditContext();
+
+	function onKeyDown(event) {
+		if (isKeyboardEvent.primary(event, 'k') || (!url && event.keyCode === ENTER)) {
+			setIsLinkOpen(true);
+		}
+	}
+
+	const link = {
+		url,
+		opensInNewTab,
+		title: value && navStripHTML(value), // don't allow HTML to display inside the <LinkControl>
+	};
+
+	const blockProps = useBlockProps({
+		ref,
+		className: classnames('wp-block-navigation-item', {
+			'is-editing': isSelected,
+			'has-link': !!url,
+		}),
+		onKeyDown,
+	});
+
+	if (!url) {
+		blockProps.onClick = () => setIsLinkOpen(true);
+	}
+
+	/**
+	 * The hook shouldn't be necessary but due to a focus loss happening
+	 * when selecting a suggestion in the link popover, we force close on block unselection.
+	 */
+	useEffect(() => {
+		if (!isSelected) {
+			setIsLinkOpen(false);
+		}
+	}, [isSelected]);
+
+	/**
+	 * Check if the URL and Value are set. If yes, then the component is valid.
+	 * Otherwise, we will output a visual reminder to the editor that one of the
+	 * two needs to be set.
+	 */
+	useEffect(() => {
+		setIsValid(!!url && !!value);
+	}, [url, value]);
+
+	return (
+		// eslint-disable-next-line react/jsx-props-no-spreading
+		<span {...blockProps}>
+			{/* eslint-disable jsx-a11y/anchor-is-valid */}
+			<LinkOutput>
```

Comment:
I wonder whether we could by without this wrapper. Essentially I'm wondering if we can get the markup to closer resemble the frontend markup. 

My main issue is that we are nesting a block level element inside of the inline anchor tag. 

We can ressolve this by either setting the `RichText` to `tagName="a"` and then add the custom styling for in in css rather than a styled component. 

I think it would also be fine to switch the `RichText` to `tagName="span"` so it at least is an inline element inside the inline link